src/components/join-flow/join-flow.jsx

```diff
@@ -121,21 +121,27 @@ class JoinFlow extends React.Component {
     //   * "birth_month": ["Ensure this value is greater than or equal to 1."]
     handleRegistrationResponse (err, body, res) {
         this.setState({
-            numAttempts: this.state.numAttempts + 1,
-            waiting: false
+            numAttempts: this.state.numAttempts + 1
         }, () => {
             const success = this.registrationIsSuccessful(err, body, res);
             if (success) {
-                this.props.refreshSession();
-                this.setState({step: this.state.step + 1});
+                this.props.refreshSessionWithRetry().then(() => {
```

Comment:
Note that -- at least for now -- we are not specifically checking to see if the user session did eventually come through. If it didn't, here we're not bailing on the registration -- we're still showing the user the Welcome step, with their username (stored locally from before).

Alternatively, we could do something else... but I'm not sure what.